crates/oxc_linter/src/rules/jest/valid_expect.rs

```diff
@@ -0,0 +1,713 @@
+                let AstKind::CallExpression(call_expr) = target_node.kind() else { return };
+                span = call_expr.span;
+                Message::AsyncMustBeAwaited.details()
+            } else {
+                let AstKind::CallExpression(call_expr) = final_node.kind() else { return };
+                span = call_expr.span;
+                Message::PromisesWithAsyncAssertionsMustBeAwaited.details()
+            };
+            ctx.diagnostic(ValidExpectDiagnostic(error, help, span));
+        }
+    }
+}
+
+fn get_parent_node_of_element<'a, 'b>(
+    element: &'b MemberExpressionElement<'a>,
+    ctx: &'b LintContext<'a>,
+) -> Option<&'b AstNode<'a>> {
+    let MemberExpressionElement::Expression(Expression::Identifier(ident)) = element else {
+        return None;
+    };
+    let reference_id = ident.reference_id.get()?;
+    let reference = ctx.symbols().get_reference(reference_id);
+    ctx.nodes().parent_node(reference.node_id())
+}
+
+fn find_top_most_member_expression<'a, 'b>(
+    node: &'b AstNode<'a>,
+    ctx: &'b LintContext<'a>,
+) -> Option<&'b MemberExpression<'a>> {
+    let mut top_most_member_expression = None;
+    let mut node = node;
+
+    loop {
+        let parent = ctx.nodes().parent_node(node.id())?;
+        match node.kind() {
+            AstKind::MemberExpression(member_expr) => {
+                top_most_member_expression = Some(member_expr);
+            }
+            _ => {
+                if !matches!(parent.kind(), AstKind::MemberExpression(_)) {
+                    break;
+                }
+            }
+        }
+        node = parent;
+    }
+
+    top_most_member_expression
+}
+
+fn is_acceptable_return_node<'a, 'b>(
+    node: &'b AstNode<'a>,
+    allow_return: bool,
+    ctx: &'b LintContext<'a>,
+) -> bool {
+    if allow_return && matches!(node.kind(), AstKind::ReturnStatement(_)) {
+        return true;
+    }
+
+    match node.kind() {
+        AstKind::ConditionalExpression(_)
+        | AstKind::Argument(_)
+        | AstKind::ExpressionStatement(_)
+        | AstKind::FunctionBody(_) => {
+            let Some(parent) = ctx.nodes().parent_node(node.id()) else { return false };
+            is_acceptable_return_node(parent, allow_return, ctx)
+        }
+        AstKind::ArrowExpression(arrow_expr) => arrow_expr.expression,
+        AstKind::AwaitExpression(_) => true,
+        _ => false,
+    }
+}
+
+fn get_parent_with_ignore<'a, 'b>(
+    node: &'b AstNode<'a>,
+    ctx: &'b LintContext<'a>,
+) -> Option<&'b AstNode<'a>> {
+    let parent = ctx.nodes().parent_node(node.id())?;
+
+    // ignore some nodes
+    if matches!(
+        parent.kind(),
+        AstKind::Argument(_)
+            | AstKind::ExpressionArrayElement(_)
+            | AstKind::ArrayExpressionElement(_)
+    ) {
+        return get_parent_with_ignore(parent, ctx);
+    }
+
+    Some(parent)
+}
+
+fn find_promise_call_expression_node<'a, 'b>(
```

Comment:
```js
Promise.all([
expect(Promise.reject(2)).rejects.not.toBeDefined(), 
expect(Promise.reject(2)).rejects.not.toBeDefined()
])
```

The  parameter `node` will be one of `expect(Promise.reject(2)).rejects.not.toBeDefined()`, and returns the node of `Promise.all`.